src/bartiq/compilation/_compile.py

```diff
@@ -191,6 +196,13 @@ def _process_repeated_resources(
             new_value = repetition.sequence_sum(resource.value, backend)
         elif resource.type == "multiplicative":
             new_value = repetition.sequence_prod(resource.value, backend)
+        elif ast.literal_eval(os.environ.get(REPETITION_ALLOW_ARBITRARY_RESOURCES_ENV, "False")):
+            new_value = resource.value
+            warnings.warn(
+                f'Can\'t process resource "{resource.name}" of type "{resource.type}" in repetitive structure.'
+                "Passing its value as is without modifications. "
+                f"To change the behaviour, set {REPETITION_ALLOW_ARBITRARY_RESOURCES_ENV} env to False."
+            )
```

Comment:
[Issue, minor non-blocking] While I agree this is the best fix in the short term, could we add an issue (and maybe a comment referencing said issue) to resolve this more thoroughly in the future (as in, properly define rules for repetition of qubits/other resource types that can handle more general repetition structures, and which can be handled without passing through _every_ other resource)?